docs/examples/use_cases/pytorch/resnet50/main.py

```diff
@@ -245,41 +266,78 @@ def resume():
         traindir = args.data[0]
         valdir= args.data[1]
 
-    if(args.arch == "inception_v3"):
+    if args.arch == "inception_v3":
         raise RuntimeError("Currently, inception_v3 is not supported by this example.")
         # crop_size = 299
         # val_size = 320 # I chose this value arbitrarily, we can adjust.
     else:
         crop_size = 224
         val_size = 256
 
-    pipe = create_dali_pipeline(batch_size=args.batch_size,
-                                num_threads=args.workers,
-                                device_id=args.local_rank,
-                                seed=12 + args.local_rank,
-                                data_dir=traindir,
-                                crop=crop_size,
-                                size=val_size,
-                                dali_cpu=args.dali_cpu,
-                                shard_id=args.local_rank,
-                                num_shards=args.world_size,
-                                is_training=True)
-    pipe.build()
-    train_loader = DALIClassificationIterator(pipe, reader_name="Reader", last_batch_policy=LastBatchPolicy.PARTIAL)
-
-    pipe = create_dali_pipeline(batch_size=args.batch_size,
-                                num_threads=args.workers,
-                                device_id=args.local_rank,
-                                seed=12 + args.local_rank,
-                                data_dir=valdir,
-                                crop=crop_size,
-                                size=val_size,
-                                dali_cpu=args.dali_cpu,
-                                shard_id=args.local_rank,
-                                num_shards=args.world_size,
-                                is_training=False)
-    pipe.build()
-    val_loader = DALIClassificationIterator(pipe, reader_name="Reader", last_batch_policy=LastBatchPolicy.PARTIAL)
+    train_loader = None
+    val_loader = None
+    if not args.disable_dali:
+        pipe = create_dali_pipeline(batch_size=args.batch_size,
+                                    num_threads=args.workers,
+                                    device_id=args.local_rank,
+                                    seed=12 + args.local_rank,
+                                    data_dir=traindir,
+                                    crop=crop_size,
+                                    size=val_size,
+                                    dali_cpu=args.dali_cpu,
+                                    shard_id=args.local_rank,
+                                    num_shards=args.world_size,
+                                    is_training=True)
+        pipe.build()
+        train_loader = DALIClassificationIterator(pipe, reader_name="Reader",
+                                                  last_batch_policy=LastBatchPolicy.PARTIAL,
+                                                  auto_reset =True)
+
+        pipe = create_dali_pipeline(batch_size=args.batch_size,
```

Comment:
Nitpick, my eyes hurt due to overwriting of the pipeline.

Reply:
Done